### ISSUE TITLE
MySQL: reporting error when setting a column to NULL

### ISSUE THREAD

**Issue by ngdah** [sqlfluff/sqlfluff#1774]
## Expected Behaviour
SQLFluff should pass lint check when setting a column to NULL using MySQL.

## Observed Behaviour
SQLFluff reported L049 error:
```
L:   1 | P:  49 | L049 | Comparisons with NULL should use "IS" or "IS NOT".
```

## Steps to Reproduce
Prepare a file `update_null.sql`:
```sql
UPDATE `foo`.`bar` SET `col` = NULL
WHERE `col` = "";

```

Run `sqlfluff lint --dialect=mysql update_null.sql`

Output will be
```
== [update_null.sql] FAIL
L:   1 | P:  30 | L049 | Comparisons with NULL should use "IS" or "IS NOT".
All Finished 📜 🎉!
```

## Dialect
MySQL

## Version
Python 3.9.7
sqlfluff, version 0.7.1

## Configuration
default
